test/e2e/suites/managed/machine_pool.go

```diff
@@ -63,11 +70,25 @@ func ManagedMachinePoolSpec(ctx context.Context, inputGetter func() ManagedMachi
 	})
 	Expect(cluster).NotTo(BeNil(), "couldn't find CAPI cluster")
 
-	shared.Byf("creating an applying the %s template", EKSManagedPoolOnlyFlavor)
+	shared.Byf("creating an applying the %s template", input.Flavor)
 	configCluster := input.ConfigClusterFn(input.ClusterName, input.Namespace.Name)
-	configCluster.Flavor = EKSManagedPoolOnlyFlavor
+	configCluster.Flavor = input.Flavor
 	configCluster.WorkerMachineCount = pointer.Int64Ptr(1)
-	err := shared.ApplyTemplate(ctx, configCluster, input.BootstrapClusterProxy)
+	workloadClusterTemplate := shared.GetTemplate(ctx, configCluster)
+	if input.UsesLaunchTemplate {
+		userDataTemplate := `#!/bin/bash
```

Comment:
Why do we need this? Does this not come from the bootstrap secret?

(i will go back and check the code ^^^^)

Reply:
Created #3363  for this